packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3648,6 +3649,20 @@ class _UpdateTextSelectionAction<T extends DirectionalCaretMovementIntent> exten
       ? TextSelection.fromPosition(newExtent)
       : textBoundarySelection.extendTo(newExtent);
 
+    // If collapseAtReversal is true and would have an effect, collapse it.
```

Comment:
@LongCatIsLooong How would you prefer that we implement this "collapseAtReversal" feature within your new Actions setup?  Feel free to butcher this if it's totally off, I'm still getting the hang of your new refactor so I've just put down a quick attempt.

Reply:
Yeah this makes sense to me!